### DISCUSSION TITLE
Colab doesn't work (reinstalled two times)

### DISCUSSION THREAD

**Discussion by ClaraSanders** [TheLastBen/fast-stable-diffusion#2078] [Q&A]
I've been using it for some hours today until now when it stopped working. I guess it was updated and something was screwed over. I reinstalled it two times and I get the same error:

ControlNet v1.1.125
ControlNet v1.1.125
Calculating sha256 for /content/gdrive/MyDrive/sd/stable-diffusion-webui/models/Stable-diffusion/mrcV1_mrcV1.ckpt: f8e22a0df8de7321b1df44dc07c1d251c84b09dd6347d1d16235bb8d386168c5
Loading weights [f8e22a0df8] from /content/gdrive/MyDrive/sd/stable-diffusion-webui/models/Stable-diffusion/mrcV1_mrcV1.ckpt
Creating model from config: /content/gdrive/MyDrive/sd/stable-diffusion-webui/configs/v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Downloading (…)olve/main/vocab.json100% 961k/961k [00:00<00:00, 1.84MB/s]
Downloading (…)olve/main/merges.txt100% 525k/525k [00:00<00:00, 26.4MB/s]
Downloading (…)cial_tokens_map.json100% 389/389 [00:00<00:00, 385kB/s]
Downloading (…)okenizer_config.json100% 905/905 [00:00<00:00, 824kB/s]
Downloading (…)lve/main/config.json100% 4.52k/4.52k [00:00<00:00, 4.53MB/s]
Applying scaled dot product cross attention optimization.
Textual inversion embeddings loaded(0): 
Model loaded in 56.9s (calculate hash: 31.1s, load weights from disk: 16.1s, create model: 4.0s, apply weights to model: 3.3s, apply half(): 1.3s, move model to device: 0.8s).
Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 337, in <module>
    webui()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 240, in webui
    shared.demo = modules.ui.create_ui();shared.demo.queue(concurrency_count=999999,status_update_rate=0.1)
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/ui.py", line 461, in create_ui
    txt2img_prompt, txt2img_prompt_styles, txt2img_negative_prompt, submit, _, _, txt2img_prompt_style_apply, txt2img_save_style, txt2img_paste, extra_networks_button, token_counter, token_button, negative_token_counter, negative_token_button = create_toprow(is_img2img=False)
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/ui.py", line 309, in create_toprow
    skip.click(
  File "/usr/local/lib/python3.10/dist-packages/gradio/events.py", line 129, in __call__
    dep, dep_index = self.trigger.set_event_trigger(
TypeError: Block.set_event_trigger() got an unexpected keyword argument 'trigger_after'

**Comment by meunumerotim333** [comment]
Use the latest update. Make another copy

**Reply by ClaraSanders** [reply]
Well, I did it two times (and tried on two different accounts), unless somebody changed the files this last hour. 
EDIT: I Tried a third time just now and same error.

**Reply by meunumerotim333** [reply]
He have updated the file. I had error too after he update. But copying a new one fixed to me. I'm using right now

**Reply by ClaraSanders** [reply]
A copy? I never copy the colab in my Drive I always go and get the latest one from Ben's.

**Reply by dilectiogames** [reply]
> A copy? I never copy the colab in my Drive I always go and get the latest one from Ben's.

that at one point blocked my google account. using it directly. it detected something strange with google drive and I needed to restore the account

**Reply by meunumerotim333** [reply]
> A copy? I never copy the colab in my Drive I always go and get the latest one from Ben's.

I always copy since last year. Use alt accounts if you don't trust or whatever. I use 19 different Gmails on profiles. Sometimes I use multiple accounts at the same time + VPN. But I always copied to drive. Every update I make a new copy.

**Comment by dilectiogames** [comment]
you don't place the stable diffusion in the normal folder. you keep them outside of the sd installation and you give it the link to the model in the notebook interface

**Reply by ClaraSanders** [reply]
This won't work for me anyway, since I use it for merging and my folder has like 10 models. How I am supposed to load them with your method?

**Reply by HorrySheet** [reply]
Im confused as to what we're supposed to do as well. I have always just used the latest colab and just recently started using the interface to load new models I find links for from civit on huggingface.  But now it wont load no matter what I do. 3 accounts.. I guess I can delete my SD folder on one of them but it looks like that's not working for everyone and I have plenty of models and loras and embeddings etc .. different stuff saved in each SD folder.. so like..    Hmm ..  Am I just to delete the SD folder and keep the WebUI folder or something? Is that what we're being asked to do here? (If that's even a thing?? - Going off my memory of what I think the folder structure is in my gdrive)  Assuming the SD folder will rebuild itself at that point and I can keep my models and whatnot intact if I delete the sd folder?? Can someone confirm plz?

**Reply by dilectiogames** [reply]
I know it sucks but google is actively messing with stable diffusion installations

trying to run that part of the notebook using a model that is in the normal folder triggered a ban warning from google.

try having a model outside and start from there then see if you can switch from the webui to others model. should work, didn't tried but should work

**Reply by dilectiogames** [reply]
> Im confused as to what we're supposed to do as well. I have always just used the latest colab and just recently started using the interface to load new models I find links for from civit on huggingface. But now it wont load no matter what I do. 3 accounts.. I guess I can delete my SD folder on one of them but it looks like that's not working for everyone and I have plenty of models and loras and embeddings etc .. different stuff saved in each SD folder.. so like.. Hmm .. Am I just to delete the SD folder and keep the WebUI folder or something? Is that what we're being asked to do here? (If that's even a thing?? - Going off my memory of what I think the folder structure is in my gdrive) Assuming the SD folder will rebuild itself at that point and I can keep my models and whatnot intact if I delete the sd folder?? Can someone confirm plz?

it gets messy, now each day I open the colab it doesn't work anymore. I've figured out how to move the essential parts out of the sd installation, usually the sd models, loras, embeddings, vae, then the controlnet models.

then you wipe the sd installation and use the newest notebook.

it works till next day when google will once again mess your installation

google don't like people using the stable diffusion but they don't ban this app directly.

**Reply by HorrySheet** [reply]
So I take it you're having to login to your google drive in a new tab and paste all your custom stuff back into the SD folder after it's rebuilt from the new colab link?  Am I thinking that through correctly?

**Reply by ClaraSanders** [reply]
I doubt it. This won´t work. I did it five times now using three different accounts: Two paid ones and one free. I deleted the sd folder and started over and I still get the same error using a clean installation without loras, embeddings, or even VAEs, just one model. If this is really Google's fault I'm gonna get really really pissed to the point that I might cancel my two subscriptions.

**Reply by HorrySheet** [reply]
Well I was just about to launch some services but I guess I'll have to put that on hold heh.  I have a damn pc that can run this but I just dont have it with me.. using colabs has really been a pain especially coming from already knowing how much nicer it is to run locally.. this drives me nuts..

**Reply by dilectiogames** [reply]
> I doubt it. This won´t work. I did it five times now using three different accounts: Two paid ones and one free. I deleted the sd folder and started over and I still get the same error using a clean installation without loras, embeddings, or even VAEs, just one model. If this is really Google's fault I'm gonna get really really pissed to the point that I might cancel my two subscriptions.

I can confirm that it worked one hour ago using latest notebook copied on the drive, using link to SD model outside of SD folder.

controlnet was installed using notebook option but without downloading any model. I've had those already in the drive and I've moved them in the correct folder (is in the extensions folder not the normal model folder) if you add the model in the wrong place notebook will break and hangup when using controlnet

but I can also confirm that I've did the same things yesterday worked for hours, then today was broken again.

so it highly depends when you install the notebook if google is actively messing with the back end at that time

**Reply by ClaraSanders** [reply]
Anyway If you can only use one model at a time and without Controlnet or extensions the colab is useless for me. I will wait until the inevitable solution I hope. Perhaps is not Google's fault there are many people complaining on Reddit about the latest Automatic update destroying their local installations.

**Reply by dilectiogames** [reply]
are you reading?

I said I've used controllnet + loras + everything

**Reply by ClaraSanders** [reply]
I don't understand. So you can access the full models' folder outside the sd folder or only ONE model at a time?

**Reply by dilectiogames** [reply]
IF you get the message about google banning you if you run the part of the colab where you set up a link to a model or download a model, then use a model that is outside the SD installation, then after the web ui has started, and you have tested if all works using the webui, then you cna switch the models using the webui interface at the top.

but only IF you get the ban message warning. at google there is no one that looks at these codes, is all automatic. I don't know why google is triggered by that piece of code but is better to not argue with it else you may lose the account

**Reply by ClaraSanders** [reply]
Thank you for trying to help me, but it still doesn't work. I load a model from outside the SD folder using a public link and it loads fine but the last cell always gives several errors.

Calculating sha256 for /content/model.ckpt: df85a9e11f8eac3f97350bf3580978c2dcd4997159655b3e88a2a9ead4abb838
Loading weights [df85a9e11f] from /content/model.ckpt
Creating model from config: /content/gdrive/MyDrive/sd/stable-diffusion-webui/configs/v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Downloading (…)olve/main/vocab.json100% 961k/961k [00:00<00:00, 24.8MB/s]
Downloading (…)olve/main/merges.txt100% 525k/525k [00:00<00:00, 22.2MB/s]
Downloading (…)cial_tokens_map.json100% 389/389 [00:00<00:00, 355kB/s]
Downloading (…)okenizer_config.json100% 905/905 [00:00<00:00, 881kB/s]
Downloading (…)lve/main/config.json100% 4.52k/4.52k [00:00<00:00, 5.69MB/s]
Applying scaled dot product cross attention optimization.
Textual inversion embeddings loaded(0): 
Model loaded in 20.2s (calculate hash: 6.0s, load weights from disk: 7.5s, create model: 3.6s, apply weights to model: 1.9s, apply half(): 0.6s, move model to device: 0.6s).
Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 413, in <module>
    webui()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 302, in webui
    shared.demo = modules.ui.create_ui();shared.demo.queue(concurrency_count=999999,status_update_rate=0.1)
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/ui.py", line 463, in create_ui
    with gr.Blocks(analytics_enabled=False) as txt2img_interface:
  File "/usr/local/lib/python3.10/dist-packages/gradio/blocks.py", line 486, in __init__
    super().__init__(render=False, **kwargs)
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/scripts.py", line 561, in BlockContext_init
    add_classes_to_gradio_component(self)
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/scripts.py", line 529, in add_classes_to_gradio_component
    comp.elem_classes = ["gradio-" + comp.get_block_name(), *(comp.elem_classes or [])]
AttributeError: 'Blocks' object has no attribute 'elem_classes'

I will wait one day or two and if this continues to fail I will cancel my subscription and I will use runpod (I have no idea how it works lol).

**Reply by dilectiogames** [reply]
I don't know what these errors are, never seen them before.

are you sure the model is correct, downloaded in full? maybe is broken

**Reply by Headupdef** [reply]
don't feel bad, mine doesn't work on google at all anymore and I simply don't understand runpod and paperspace always says the  selected gpu is busy

**Reply by TheLastBen** [reply]
> don't feel bad, mine doesn't work on google at all anymore and I simply don't understand runpod and paperspace always says the selected gpu is busy

https://www.runpod.io/console/gpu-browse?template=runpod-stable-unified here you have a big list of gpu at good prices

**Reply by dilectiogames** [reply]
no need for google colab anymore if you have a less powerful GPU at home. Use this solution to render images without getting into CUDA out of memory issues

https://www.youtube.com/watch?v=EmA0RwWv-os
